src/test/java/com/crossover/trial/journals/rest/JournalServiceTest.java

```diff
@@ -0,0 +1,162 @@
+package com.crossover.trial.journals.rest;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.fail;
+
+import java.util.List;
+import java.util.Optional;
+
+import com.crossover.trial.journals.Application;
+import com.crossover.trial.journals.model.Journal;
+import com.crossover.trial.journals.model.Publisher;
+import com.crossover.trial.journals.model.User;
+import com.crossover.trial.journals.repository.PublisherRepository;
+import com.crossover.trial.journals.service.JournalService;
+import com.crossover.trial.journals.service.ServiceException;
+import com.crossover.trial.journals.service.UserService;
+import org.junit.FixMethodOrder;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.MethodSorters;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.SpringApplicationConfiguration;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+import org.springframework.test.context.web.WebAppConfiguration;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringApplicationConfiguration(classes = Application.class)
+@WebAppConfiguration
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class JournalServiceTest {
```

Comment:
It would be better imo to write test in an behavioural driven test. 

Instead of methods like `#publishFail`, which doesn't describe the test scenario. It would have been better to write method name with `#shouldThrowServiceExceptionForInvalidId`.